src/aero_kernel/src/syscall/ipc.rs

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2021-2022 The Aero Project Developers.
+ *
+ * This file is part of The Aero Project.
+ *
+ * Aero is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Aero is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Aero. If not, see <https://www.gnu.org/licenses/>.
+ */
+use crate::{
+    userland::scheduler::get_scheduler,
+    utils::{
+        sync::{BlockQueue, Mutex},
+        validate_slice_mut,
+    },
+};
+use aero_syscall::AeroSyscallError;
+use alloc::{collections::VecDeque, vec::Vec};
+use hashbrown::HashMap;
+use lazy_static::lazy_static;
+
+struct Message {
+    from: usize,
+    data: Vec<u8>,
+}
+struct MessageQueue {
+    pub queue: VecDeque<Message>,
+}
+
+lazy_static! {
+    static ref BLOCK_QUEUE: BlockQueue = BlockQueue::new();
+    static ref MESSAGES: Mutex<HashMap<usize, MessageQueue>> = Mutex::new(HashMap::new());
```

Comment:
Instead of storing the message queue in a global hashmap, store it in the task struct itself. It will make stuff much easier.

Reply:
yep done